core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -181,6 +181,7 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
   this.serverConfig.setProperty(KafkaConfig.DefaultReplicationFactorProp, "3")
   this.serverConfig.setProperty(KafkaConfig.ConnectionsMaxReauthMsProp, "1500")
   this.consumerConfig.setProperty(ConsumerConfig.GROUP_ID_CONFIG, "group")
+  this.consumerConfig.setProperty(ConsumerConfig.METADATA_MAX_AGE_CONFIG, "1500")
```

Comment:
This is a fix for the test flakiness itself: when the metadata has an error, it would not be requested update immediately but still wait on its metadata age to expire; so if we are unlucky and still get this error when assigned with tp1/tp2, we are doomed to fail since the metadata would not be refreshed within the 10 sec timeout.

Reply:
The metadata refresh policy seems a bit broken if it can go 5 minutes between updating the metadata due to an error :(

Reply:
From my observation, it would still if there's a fatal error like `AuthorizationException`, in this test we actually are relying on manual `assign` which would always request a metadata update, but if that update still get the `AuthorizationException` on some of the topic metadata it would wait for another refresh period.

Reply:
AuthorizationException should not happen during the test, though, right?  We choose to treat AuthorizationException differently than transitory errors specifically because we believe authorization failures should be permanent (in most cases....)

Reply:
I actually see it being thrown from the metadata, in `metadata.maybeThrowAnyException()` inside `poll`. The problem is that, within this test, if this race condition happens:

1. We update ACL for `topic`.
2. We change `consumer.assign`, which would automatically trigger a metadata request.
3. The metadata still have the AuthorizationException even after 1) since it is not propagated yet.

Then after it is thrown the metadata would no longer be updated until 5 min later, whereas we only wait for 10 secs.

Reply:
Could we wait for the metadata propagation before calling assign?

Reply:
For authorization errors, it seems reasonable not to retry too eagerly. We do have logic to retry immediately if the error is invalid metadata (e.g. NOT_LEADER).

Reply:
@ijuma actually my previous statement is not correct, what we actually did is:

1. Change `consumer.assign`, and then check that produce to other authorized topic is okay but fetching will throw.
2. Then update the ACL for `topic`.

So there are two scenarios when 1) throw: it can either throw from the update-position call where the metadata is not updated, or it can throw from the already-requested metadata that still have the `TopicAuthorizationException`. In the latter case since the metadata is already updated it would not be updated again until the metadata.age elapsed.

Reply:
I see. OK, maybe what you've done is easiest then. 1.5 seconds seems a bit slow still, should it be even lower?

Reply:
I guess I can make that even lower, the previous value is just to resolve the flakiness.